### DISCUSSION TITLE
How to retrieve database username and Password via api?

### DISCUSSION THREAD

**Discussion by naresh-sanda** [planetscale/discussion#421] [FAQs] [answered]
Hello Squad, 

I want to create a DB and knows its assigned username and password and other connection details.
Please let mw know the API  , in create Db I expected the same but not receiving them as part of the response.

Tried Get a password API : but returned with 404 

https://api.planetscale.com/v1/organizations/naresh-xxxx/databases/sample_xxxx/branches/main/passwords?database=sample_xxxx&branch=main&organization=naresh-xxxx

Thanks , 
Naresh

**Comment by jgreet** [comment, marked as answer]
Password are created separately from databases, and scoped to branches. So this would be two calls:

First, a POST to `v1/organizations/{organization}/databases`  to create the database.
  - https://api-docs.planetscale.com/reference/create-a-database

And then a POST to `v1/organizations/{organization}/databases/{database}/branches/main/passwords` to create a connection for the main branch of the new database. The text of the password is only available at the time it's created.
  - https://api-docs.planetscale.com/reference/create-a-password

**Reply by naresh-sanda** [reply]
Thank you jgreet.
